docker-compose.yml

```diff
@@ -29,6 +29,35 @@ services:
       NUM_THREADS: 8 # default value for NUM_THREADS
       VELOX_DEPENDENCY_SOURCE: BUNDLED # Build dependencies from source
       CCACHE_DIR: "/velox/.ccache"
+      CMAKE_EXPORT_COMPILE_COMMANDS: 1
+    volumes:
+      - .:/velox:delegated
+    command: scripts/docker-command.sh
+
+  ubuntu-cuda-cpp:
+    # Usage:
+    #   docker-compose pull ubuntu-cuda-cpp or docker-compose build ubuntu-cuda-cpp
+    #   docker-compose run --rm ubuntu-cuda-cpp
+    #   or
+    #   docker-compose run -e NUM_THREADS=<NUMBER_OF_THREADS_TO_USE> --rm ubuntu-cuda-cpp
+    #   to set the number of threads used during compilation
+    image: ghcr.io/facebookincubator/velox-dev:ubuntu-22.04-cuda-12.8
+    build:
+      context: .
+      dockerfile: scripts/ubuntu-22.04-cuda-12.8-cpp.dockerfile
+    environment:
+      NUM_THREADS: 8 # default value for NUM_THREADS
+      VELOX_DEPENDENCY_SOURCE: BUNDLED # Build dependencies from source
```

Comment:
If we are building an image here, would be better to take advantage of that and have dependencies be AUTO/SYSTEM (to not add to the build times) ? (any concerns with that @assignUser )